ui/src/main/java/com/dat3m/ui/result/ReachabilityResult.java

```diff
@@ -114,18 +114,18 @@ private void run(){
         }
     }
 
-    private void buildVerdict(Result result, Model m, SolverContext ctx){
+    private void buildVerdict(Result result, ProverEnvironment prover, SolverContext ctx) throws SolverException {
         StringBuilder sb = new StringBuilder();
         if(result.equals(FAIL)) {
     		for(Axiom ax : wmm.getAxioms()) {
-        		if(ax.isFlagged() && TRUE.equals(m.evaluate(CAT.getSMTVariable(ax, ctx)))) {
+        		if(ax.isFlagged() && TRUE.equals(prover.getModel().evaluate(CAT.getSMTVariable(ax, ctx)))) {
         			sb.append("Flag " + (ax.getName() != null ? ax.getName() : ax.getRelation().getName())).append("\n");
         		}
     		}
         }
 		// TODO We might want to output different messages once we allow to check LIVENESS from the UI
 		sb.append("Condition ").append(program.getAss().toStringWithType()).append("\n");
-		sb.append(program.getFormat().equals(LITMUS) ? (result.equals(FAIL) && TRUE.equals(m.evaluate(REACHABILITY.getSMTVariable(ctx)))) ? "Ok" : "No" : result).append("\n");
+		sb.append(program.getFormat().equals(LITMUS) ? (result.equals(FAIL) && TRUE.equals(prover.getModel().evaluate(REACHABILITY.getSMTVariable(ctx)))) ? "Ok" : "No" : result).append("\n");
```

Comment:
I would make sure to call `prover.getModel` only once and store it in a variable (or null, if there is model).
It is not clear if the underlying solver reconstructs the model on each call.

Also, models should get closed properly (`model.close`).